[BUG] [java-camel] pom.xml is not generated by default despite the documentation

#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

<!--
Please follow the issue template below for bug reports.
Also please indicate in the issue title which language/library is concerned. E.g.:  [BUG][JAVA] Bug generating foo with bar 
-->

##### Description

<!-- describe what is the question, suggestion or issue and why this is a problem for you. -->

The java-camel generator doesn't seem to fully reflect its documentation regarding generating pom.xml.

1. Running the generator without any options doesn't generate pom.xml, as follows:

```
$ curl -sLO https://repo1.maven.org/maven2/org/openapitools/openapi-generator-cli/7.19.0/openapi-generator-cli-7.19.0.jar -O openapi-generator-cli.jar
$ curl -sLO https://raw.githubusercontent.com/OpenAPITools/openapi-generator/refs/tags/v7.19.0/modules/openapi-generator/src/test/resources/3_1/petstore.yaml
$ java -jar openapi-generator-cli-7.19.0.jar generate -g java-camel -i petstore.yaml -o java-camel-petstore
$ ls -al java-camel-petstore
total 20
drwxrwxr-x  4 sekikn sekikn 4096 Feb  3 18:16 .
drwxr-x--- 90 sekikn sekikn 4096 Feb  3 18:16 ..
drwxrwxr-x  2 sekikn sekikn 4096 Feb  3 18:16 .openapi-generator
-rw-rw-r--  1 sekikn sekikn 1040 Feb  3 18:16 .openapi-generator-ignore
drwxrwxr-x  3 sekikn sekikn 4096 Feb  3 18:16 src
```

[That file is generated only if the value of the library option is `spring-boot`](https://github.com/OpenAPITools/openapi-generator/blob/v7.19.0/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaCamelServerCodegen.java#L115-L127). And [according to the document](https://github.com/OpenAPITools/openapi-generator/blob/v7.19.0/docs/generators/java-camel.md?plain=1#L73), its default value is `spring-boot`. So it should be generated even if the option is omitted.

2. If specifying `library=spring-boot` explicitly, pom.xml is created:

```
$ java -jar openapi-generator-cli-7.19.0.jar generate -g java-camel -i petstore.yaml -o java-camel-petstore --additional-properties=library=spring-boot
$ cd java-camel-petstore
$ ls -al
total 32
drwxrwxr-x  4 sekikn sekikn 4096 Feb  3 18:29 .
drwxr-x--- 90 sekikn sekikn 4096 Feb  3 18:29 ..
drwxrwxr-x  2 sekikn sekikn 4096 Feb  3 18:29 .openapi-generator
-rw-rw-r--  1 sekikn sekikn 1040 Feb  3 18:29 .openapi-generator-ignore
-rw-rw-r--  1 sekikn sekikn 6076 Feb  3 18:29 pom.xml
-rw-rw-r--  1 sekikn sekikn   75 Feb  3 18:29 README.md
drwxrwxr-x  4 sekikn sekikn 4096 Feb  3 18:29 src
```

But building that project fails due to the lack of the io.swagger.core.v3:swagger-annotations library, as follows:

```
$ mvn clean install

(snip)

[INFO] ------------------------------------------------------------------------                                                                                                               
[INFO] BUILD FAILURE                                                                                                                                                                          
[INFO] ------------------------------------------------------------------------                                                                                                               
[INFO] Total time:  1.175 s                                                                                                                                                                   
[INFO] Finished at: 2026-02-03T18:30:19+09:00                                                                                                                                                 
[INFO] ------------------------------------------------------------------------                                                                                                               
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.10.1:compile (default-compile) on project openapi-camel: Compilation failure: Compilation failure:            
[ERROR] /home/sekikn/java-camel-petstore/src/main/java/org/openapitools/model/Pet.java:[17,43] package io.swagger.v3.oas.annotations.media does not exist
```

[According to the document](https://github.com/OpenAPITools/openapi-generator/blob/v7.19.0/docs/generators/java-camel.md?plain=1#L25), the default value of the `annotationLibrary` option is `swagger2`, so the dependency should be provided by default as well.

##### openapi-generator version

<!-- which version of openapi-generator are you using, is it a regression? -->

v7.19.0 and the latest master

##### OpenAPI declaration file content or url

<!-- if it is a bug, a json or yaml that produces it.
If you post the code inline, please wrap it with
```yaml
(here your code)
```
(for YAML code) or
```json
(here your code)
```
(for JSON code), so it becomes more readable. If it is longer than about ten lines,
please create a Gist (https://gist.github.com) or upload it somewhere else and
link it here.
  -->

https://github.com/OpenAPITools/openapi-generator/blob/v7.19.0/modules/openapi-generator/src/test/resources/3_1/petstore.yaml

##### Generation Details

<!-- 
    Prefer CLI steps, including the language, libraries and various options. 
    Providing config-based settings allows for simpler testing across CLI and plugins. 
    For examples, see https://github.com/OpenAPITools/openapi-generator/tree/master/bin/configs
-->

See the description above

##### Steps to reproduce

<!-- unambiguous set of steps to reproduce the bug.-->

See the description above

##### Related issues/PRs

<!-- has a similar issue/PR been reported/opened before? Please do a search in https://github.com/openapitools/openapi-generator/issues?utf8=%E2%9C%93&q=is%3Aissue%20 -->

##### Suggest a fix

<!-- if you can't fix the bug yourself, perhaps you can point to what might be
  causing the problem (line of code or commit), or simply make a suggestion -->

--- a/bin/configs/java-camel-petstore-new.yaml
+++ b/bin/configs/java-camel-petstore-new.yaml
@@ -3,13 +3,11 @@ outputDir: samples/server/petstore/java-camel
 inputSpec: modules/openapi-generator/src/test/resources/3_0/petstore.yaml
 templateDir: modules/openapi-generator/src/main/resources/java-camel-server
 additionalProperties:
-  oas3: "true"
   hideGenerationTimestamp: true
   camelRestBindingMode: "auto"
   performBeanValidation: true
   #dateLibrary: "java8-localdatetime"
   camelDataformatProperties: "json.out.disableFeatures=WRITE_DATES_AS_TIMESTAMPS"
-  library: "spring-boot"
   withXml: true
   jackson: true
   camelUseDefaultValidationErrorProcessor: true

--- a/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaCamelServerCodegen.java
+++ b/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaCamelServerCodegen.java
@@ -72,7 +72,7 @@ public class JavaCamelServerCodegen extends SpringCodegen implements BeanValidat
         templateDir = "java-camel-server";
         addCliOptions();
         artifactId = "openapi-camel";
-        super.library = "";
+        annotationLibrary = AnnotationLibrary.SWAGGER2;
     }
 
     @Override

--- a/modules/openapi-generator/src/main/resources/java-camel-server/pom.mustache
+++ b/modules/openapi-generator/src/main/resources/java-camel-server/pom.mustache
@@ -118,18 +118,20 @@ Do not edit the class manually.
             <artifactId>jackson-databind-nullable</artifactId>
             <version>0.2.8</version>
         </dependency>
+        {{#swagger1AnnotationLibrary}}
         <dependency>
             <groupId>io.swagger</groupId>
             <artifactId>swagger-annotations</artifactId>
             <version>1.6.3</version>
         </dependency>
-        {{#oas3}}
+        {{/swagger1AnnotationLibrary}}
+        {{#swagger2AnnotationLibrary}}
         <dependency>
             <groupId>io.swagger.core.v3</groupId>
             <artifactId>swagger-annotations</artifactId>
             <version>2.2.8</version>
         </dependency>
-        {{/oas3}}
+        {{/swagger2AnnotationLibrary}}
         {{#jackson}}
         <dependency>
             <groupId>com.fasterxml.jackson.datatype</groupId>

--- a/samples/server/petstore/java-camel/pom.xml
+++ b/samples/server/petstore/java-camel/pom.xml
@@ -119,11 +119,6 @@ Do not edit the class manually.
             <version>0.2.8</version>
         </dependency>
         <dependency>
-            <groupId>io.swagger</groupId>
-            <artifactId>swagger-annotations</artifactId>
-            <version>1.6.3</version>
-        </dependency>
-        <dependency>
             <groupId>io.swagger.core.v3</groupId>
             <artifactId>swagger-annotations</artifactId>
             <version>2.2.8</version>

